Lib/test/test_fractions.py

```diff
@@ -117,6 +117,12 @@ def assertTypedEquals(self, expected, actual):
         self.assertEqual(type(expected), type(actual))
         self.assertEqual(expected, actual)
 
+    def assertTypedTupleEquals(self, expected, actual):
+        """Asserts that both the types and values in the tuples are the same."""
+        self.assertIsInstance(actual, tuple)
+        self.assertEqual(list(map(type, expected)), list(map(type, actual)))
```

Comment:
This is not needed. Instead add
```python
for a, e in zip(actual, expected):
    self.assertTypedEquals(a, e)
```
after the next line.

Reply:
I would then also have to check that both tuples have the same length. Honestly, I prefer the current approach, or, in fact, the `assertTupleEqual()` and `assertListEqual()` methods, which I now switched to.